internal/background/availability_queue.go

```diff
@@ -2,20 +2,29 @@ package background
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"time"
 
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
 	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
 	"github.com/RHEnVision/provisioning-backend/internal/kafka"
+	"github.com/RHEnVision/provisioning-backend/internal/metrics"
 )
 
+type availabilityRequestInfo struct {
+	identity string
+	asm      *kafka.AvailabilityStatusMessage
+}
+
 // buffered channel for incoming requests
-var kafkaAvailabilityRequest = make(chan *kafka.GenericMessage, availabilityStatusBatchSize)
+var kafkaAvailabilityRequest = make(chan availabilityRequestInfo, availabilityStatusBatchSize)
 
 // EnqueueAvailabilityStatusRequest prepares a status request check to be sent in the next
 // batch to the platform kafka. Messages can be delayed up to several seconds until sent.
 // The function can block if the enqueueing channel is full.
-func EnqueueAvailabilityStatusRequest(msg *kafka.GenericMessage) {
-	kafkaAvailabilityRequest <- msg
+func EnqueueAvailabilityStatusRequest(identity string, asm *kafka.AvailabilityStatusMessage) {
```

Comment:
Why aren't we passing ctx and getting the identity from context here in the function? I feel like it would hide bit of a complexity from the consumer (the service)

Reply:
sure, I thought it was better to pass the identity according to a similar discussion I had with @lzap  regarding passing the identity/context through a channel. 
If you think it is better to pass the context, ill change it :)

Reply:
I am thinking passing ctx to the method, method pulls the identity and passes identity to the channel.
I'm 95% behind not sending whole context through channels :)

Reply:
But in the future we might want to send the identity in different format and if we form it in this file, it's local and much easier to deal with :)

Reply:
OH sorry! I completely missed your point! of course, we should send the context in the function! 
I thought you meant to save the context in the struct 
It was a mistake passing the identity xD, fixing, thank you for noticing

Reply:
Yeah in general, context is great as the first function argument to be passed from call to call. However saving it to database, serializing to struct. sending over channel or over network or saving to file - that is not what you generally want to do.

In that case, you want to save the individual values from context and then create a new context when you receive the value. This is what we do in jobs for example.

There are cases there this might be an overkill, for example when you have 4 goroutines which communicates via channels - the availability check pipeline. But this function is our own API and it should be consistent, so here it should definitely be a context I think.